### PR TITLE
py: fix access token parsing

### LINKED ISSUE #340: TypeError in get_registered_model function when using model registry python client v0.2.5a1
**Describe the bug**

I have tried using the python client on a Jupyter notebook hosted inside the cluster, the function get_registered_model fails with the following error "TypeError: can only concatenate str (not "bytes") to str". I have checked the pod connectivity using the requests library and the model registry responds correctly.

**To Reproduce**

1. !pip install --pre model-registry=="0.2.5a1"
2. from model_registry import ModelRegistry
3. registry = ModelRegistry(
        server_address="http://test.namespace.svc.cluster.local",
        port=8080,
        author="John Doe",
        is_secure=False (True or False makes no difference)
    )
4. model = registry.get_registered_model("X")

**Expected behavior**

{'createTimeSinceEpoch': '1725286288951', 'customProperties': {}, 'id': '3', 'lastUpdateTimeSinceEpoch': '1725286288951', 'name': 'X', 'owner': 'apraglio', 'state': 'LIVE'}

**Additional context**

Error trace:

```
File /opt/app-root/lib64/python3.9/site-packages/mr_openapi/configuration.py:357, in Configuration.auth_settings(self)

350 auth = {}
351 if self.access_token is not None:
352 auth["Bearer"] = {
353 "type": "bearer",
354 "in": "header",
355 "format": "JWT",
356 "key": "Authorization", -->
357 "value": "Bearer " + self.access_token, <- affected line
358 }
359 return auth

TypeError: can only concatenate str (not "bytes") to str
```

### PR DESCRIPTION
**Pull request by Al-Pragliola**
## Description

fixes #340

After running `registry.get_registered_model("X")` without providing an user_token in ModelRegistry initialization, I got the following error:

```
File /opt/app-root/lib64/python3.9/site-packages/mr_openapi/configuration.py:357, in Configuration.auth_settings(self)

350 auth = {}
351 if self.access_token is not None:
352 auth["Bearer"] = {
353 "type": "bearer",
354 "in": "header",
355 "format": "JWT",
356 "key": "Authorization", -->
357 "value": "Bearer " + self.access_token, <- affected line
358 }
359 return auth

TypeError: can only concatenate str (not "bytes") to str
```

thanks to @isinyaaa , we figured out that there was a bug in the token parsing function in the client code and switched from reading it from "bytes mode" to "text mode"

## How Has This Been Tested?

Uploaded the wheel on the notebook and rerun.

## Merge criteria:
- All the commits have been [_signed-off_](https://github.com/kubeflow/community/tree/master/dco-signoff-hook#signing-off-commits)  (To pass the `DCO` check)

- [ ] The commits have meaningful messages; the author will squash them after approval or in case of manual merges will ask to merge with squash.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work.
- [ ] Code changes follow the [kubeflow contribution guidelines](https://www.kubeflow.org/docs/about/contributing/).

### CODE REVIEW

**Comment by tarilabs** [comment]
nice catch, thanks @Al-Pragliola @isinyaaa ;

/lgtm

just before merging, can you comment on potential idea to create a non-regression testing for this?

It doesn't "have to be" in this PR, but I want to track it before we merge, thanks!

**Comment by Al-Pragliola** [comment]
What I would I do is a test where there's no user_token defined when initializing the client (ModelRegistry), `KF_PIPELINES_SA_TOKEN_PATH` env var set to a tmp file path, where the content is a mocked token, register a new model and then trying `get_registered_model` to get it back

File: clients/python/tests/conftest.py [hunk 1]
```diff
@@ -133,3 +134,19 @@ def event_loop():
 @cleanup
 def client() -> ModelRegistry:
     return ModelRegistry(REGISTRY_HOST, REGISTRY_PORT, author="author", is_secure=False)
+
+@pytest.fixture(scope="module")
+def setup_env_user_token():
+    token_file = tempfile.NamedTemporaryFile(delete=False)
+    token_file.write(b"Token")
+    token_file.close()
```
Comment (tarilabs) [inline clients/python/tests/conftest.py:142]:
```suggestion
    with tempfile.NamedTemporaryFile(delete=False) as token_file:
        token_file.write(b"Token")
```

possibly more pythonic

**Review by tarilabs — COMMENTED** [review]
Awesome to have a non-regression along with the fix, thank you so much @Al-Pragliola ! 🚀 

One small suggestion, below.

The fixture looks to me following same pattern/style than other fixtures, and looks good to me.

Any further comment @isinyaaa ?
/lgtm

File: clients/python/tests/conftest.py [hunk 2]
```diff
@@ -133,3 +134,18 @@ def event_loop():
...
+        token_file.write(b"Token")
+    old_token_path = os.getenv("KF_PIPELINES_SA_TOKEN_PATH")
+    os.environ["KF_PIPELINES_SA_TOKEN_PATH"] = token_file.name
+
+    yield token_file.name
+
+    if old_token_path is None:
+        del os.environ["KF_PIPELINES_SA_TOKEN_PATH"]
+    else:
+        os.environ["KF_PIPELINES_SA_TOKEN_PATH"] = old_token_path
```
Comment (isinyaaa) [inline clients/python/tests/conftest.py:150]: we don't actually need this, the envvar will be preserved outside the tests.

**Review by isinyaaa — APPROVED** [review]
lgtm, just a nit inline :)

File: clients/python/tests/regression_test.py [hunk 3]
```diff
@@ -22,3 +22,25 @@ def test_create_tagged_version(client: ModelRegistry):
     mv = client.get_model_version(name, version)
     assert mv
     assert mv.id
+
+@pytest.mark.e2e
+def test_get_model_without_user_token(setup_env_user_token, client):
```
Comment (isinyaaa) [inline clients/python/tests/regression_test.py:27]: I see here you're already using it. So the scope reduction is a better practice.

**Review by isinyaaa — CHANGES_REQUESTED** [review]
I had misunderstood your intent before, sorry for the noise.

**Review by isinyaaa — APPROVED** [review]
On a third look this seems fine :) -- thanks @tarilabs for the reference:
https://stackoverflow.com/questions/25660064/in-which-order-are-pytest-fixtures-executed/64471427#64471427:~:text=while%20discouraging%20from%20relying%20on%20another%20factors%20(like%20order%20in%20test%20function%20arguments)%3A

File: clients/python/tests/conftest.py [hunk 4]
```diff
@@ -133,3 +134,18 @@ def event_loop():
 @cleanup
 def client() -> ModelRegistry:
     return ModelRegistry(REGISTRY_HOST, REGISTRY_PORT, author="author", is_secure=False)
+
+@pytest.fixture(scope="module")
+def setup_env_user_token():
```
Comment (tarilabs) [inline clients/python/tests/conftest.py:139]: i.e. we rely on scope for fixture ordering, and not fn arg ordering in the actual test fn.

this fixture is to be consumed by the pytest fn as an argument, when the test wants to opt-in to the (mocked) token.
when that happens, this fixture here needs to be balled before the fixture producing the MR python client.
this is achieved by this fixture having wider scope

**Review by tarilabs — APPROVED** [review]
/lgtm
/approve

**Pull request merged**